posthog-core/src/index.ts

```diff
@@ -953,10 +953,17 @@ export abstract class PostHogCore extends PostHogCoreStateless {
         this.groups(properties.$groups)
       }
 
+      // promote $set and $set_once to top level
+      const userPropsOnce = properties?.$set_once
+      delete properties?.$set_once
+
+      // if no $set is provided we assume all properties are $set
+      const userProps = properties?.$set || properties
```

Comment:
Hacky workaround to keep back-compatibility, ideally those 2 Dicts were additional method parameters instead of a single parameter typed as a generic key/value that can be anything.